dhis-2/pom.xml

```diff
@@ -672,29 +672,23 @@
           <configuration>
             <failOnWarning>true</failOnWarning>
             <ignoredUnusedDeclaredDependencies>
-              <ignoredUnusedDeclaredDependency>org.projectlombok:lombok</ignoredUnusedDeclaredDependency>
               <ignoredUnusedDeclaredDependency>org.junit.jupiter:junit-jupiter</ignoredUnusedDeclaredDependency>
-              <ignoredUnusedDeclaredDependency>org.springframework.security:spring-security-core</ignoredUnusedDeclaredDependency>
               <ignoredUnusedDeclaredDependency>net.sf.jasperreports:jasperreports</ignoredUnusedDeclaredDependency>
               <ignoredUnusedDeclaredDependency>org.mockito:mockito-inline</ignoredUnusedDeclaredDependency>
               <ignoredUnusedDeclaredDependency>org.apache.logging.log4j:log4j-slf4j-impl</ignoredUnusedDeclaredDependency>
+              <ignoredUnusedDeclaredDependency>javax.servlet:javax.servlet-api</ignoredUnusedDeclaredDependency>
+              <ignoredUnusedDeclaredDependency>org.apache.jclouds.provider:aws-s3</ignoredUnusedDeclaredDependency>
```

Comment:
`dhis-service-core` seems to be the one declaring most jclouds dependencies. I think it would be great if we could move the dependencies that are specific to individual modules inside those modules. So we don't add these ignore configurations globally. This should lead to cleaner poms over time.

Adding a `maven-dependency-plugin` config in `dhis-service-core` with this attribute
https://www.baeldung.com/maven-plugin-override-parent#2-append-parent-configuration
might work

Reply:
I agree with the idea. I just think that before that we should really validate that such dependencies belong in the module where they are declared.
Now it is difficult to really understand it because they are only needed at runtime.